packages/did/src/Did.utils.ts

```diff
@@ -135,29 +135,35 @@ export function validateUri(
     throw new TypeError(`DID string expected, got ${typeof input}`)
   }
   const { address, fragment } = parse(input as DidUri)
-  switch (expectType) {
-    // for backwards compatibility with previous implementations, `false` maps to `Did` while `true` maps to `undefined`.
-    // @ts-ignore
-    case false:
-    case 'Did':
-      if (fragment)
-        throw new SDKErrors.DidError(
-          'Expected a Kilt DidUri but got a DidResourceUri (containing a #fragment)'
-        )
-      break
-    case 'ResourceUri':
-      if (!fragment)
-        throw new SDKErrors.DidError(
-          'Expected a Kilt DidResourceUri (containing a #fragment) but got a DidUri'
-        )
-      break
-    default:
-      break
+
+  if (
+    fragment &&
+    (expectType === 'Did' ||
+      // for backwards compatibility with previous implementations, `false` maps to `Did` while `true` maps to `undefined`.
+      (typeof expectType === 'boolean' && expectType === false))
```

Comment:
The switch statements cause lots of issues, like here only a ts-ignore could help to silence the error, while a common if statement can be written without overrides.